src/lib/tableland-calls.ts

```diff
@@ -83,54 +66,53 @@ async function GeneralizedRPC(
   };
 }
 
-export async function checkAuthorizedList(this: Connection): Promise<boolean> {
-  const authorized: boolean = await SendCall.call(
-    this,
-    await GeneralizedRPC.call(this, "authorize")
-  ).then((r) => {
-    return r.status === 200;
-  });
-  return authorized;
-}
-
 async function hash(
   this: Connection,
   query: string
 ): Promise<StructureHashReceipt> {
   const message = await GeneralizedRPC.call(this, "validateCreateTable", {
-    createStatement: query,
+    create_statement: query,
   });
+  const json = await SendCall.call(this, message);
+
+  return camelCaseKeys(json.result);
+}
 
-  const response = await SendCall.call(this, message);
-  const json = await sendResponse(response);
+async function read(this: Connection, query: string): Promise<ReadQueryResult> {
+  const message = await GeneralizedRPC.call(this, "runReadQuery", {
+    statement: query,
+  });
+  const json = await SendCall.call(this, message);
 
-  return json as StructureHashReceipt;
+  return camelCaseKeys(json.result.data);
 }
 
-async function query(
+// Note: This method returns right away, once the write request has been sent to a validator for
+//       writing to the Tableland smart contract. However, the write is not confirmed until a validator
+//       has picked up the write event from the smart contract, and digested the event locally.
+async function write(
   this: Connection,
   query: string
-): Promise<ReadQueryResult | null> {
-  const message = await GeneralizedRPC.call(this, "runSQL", {
+): Promise<WriteQueryResult> {
+  const message = await GeneralizedRPC.call(this, "relayWriteQuery", {
     statement: query,
   });
-  const response = await SendCall.call(this, message);
-  const json = await sendResponse(response);
+  const json = await SendCall.call(this, message);
 
-  return json as ReadQueryResult;
+  return camelCaseKeys(json.result.tx);
 }
 
 async function receipt(
   this: Connection,
   txnHash: string
-): Promise<ReceiptResult> {
+): Promise<ReceiptResult | void> {
   const message = await GeneralizedRPC.call(this, "getReceipt", {
-    txnHash,
+    txn_hash: txnHash,
   });
-  const response = await SendCall.call(this, message);
-  const json = await sendResponse(response);
+  const json = await SendCall.call(this, message);
 
-  return json as ReceiptResult;
+  if (json.result.receipt) return camelCaseKeys(json.result.receipt);
+  // if the transaction hasn't been digested return undefined
 }
 
```

Comment:
Not sure why we were exporting `TableMetadata`, but we don't need to do that anymore